tests/azure/subnet_test.go

```diff
@@ -0,0 +1,48 @@
+package azure_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/NeowayLabs/klb/tests/lib/azure"
+	"github.com/NeowayLabs/klb/tests/lib/azure/fixture"
+)
+
+func genSubnetName() string {
+	return fmt.Sprintf("klb-subnet-tests-%d", rand.Intn(1000))
+}
+
+func testSubnetCreate(t *testing.T, f fixture.F) {
+	nsg := genNsgName()
+	f.Shell.Run(
+		"./testdata/create_nsg.sh",
+		nsg,
+		f.ResGroupName,
+		f.Location,
+	)
+	vnet := genVnetName()
+	f.Shell.Run(
+		"./testdata/create_vnet.sh",
+		vnet,
+		f.ResGroupName,
+		f.Location,
+		"10.116.0.0/16",
+	)
+	subnet := genSubnetName()
+	f.Shell.Run(
+		"./testdata/create_subnet.sh",
+		subnet,
+		f.ResGroupName,
+		vnet,
+		"10.116.1.0/24",
```

Comment:
We need to start testing that resources are created with the right parameters instead of just checking them if they exist, this made sense with the resource group but will leave the rest of the tests anemic.